plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -195,21 +195,23 @@ fu_plugin_thunderbolt_is_native (GUdevDevice *udevice, gboolean *is_native, GErr
 {
 	g_autoptr(GFile) nvmem = NULL;
 	g_autoptr(GBytes) controller_fw = NULL;
-	gchar *content;
-	gsize length;
+	g_autoptr(GInputStream) istr = NULL;
 
 	nvmem = fu_plugin_thunderbolt_find_nvmem (udevice, TRUE, error);
 	if (nvmem == NULL)
 		return FALSE;
 
-	if (!g_file_load_contents (nvmem, NULL, &content, &length, NULL, error))
+	/* read 128 bytes, so two 64 byte reads over the Thunderbolt fabric */
+	istr = G_INPUT_STREAM (g_file_read (nvmem, NULL, error));
+	if (istr == NULL)
+		return FALSE;
+	controller_fw = g_input_stream_read_bytes (istr, 128, NULL, error);
```

Comment:
128 is a bit of a magical number since the offset for native is fixed thus far. I think that should be clarified here either with a define or mentioned in the near by comment

Reply:
Something like this, I guess (unless you prefer bitwise operations):
#define OFFSET 0x7B
#define CHUNK_SIZE 64
#define LENGTH (((OFFSET + CHUNK_SIZE - 1) / CHUNK_SIZE) * CHUNK_SIZE)

Reply:
(But OFFSET may have a better name so it can be included in fu-thunderbolt-image.h to be used both here and in fu-thunderbolt-image.c)

Reply:
@YehezkelShB what's the name of the byte in the tbt specs? FU_TBT_OFFSET_IS_NATIVE isn't right I guess because it has a bitmask.

Reply:
I don't think I ever knew anything more about this byte, or at least I can't remember it. Maybe @westeri knows more. Anyway, I think this is a reasonable name for our needs, as this is the only info we take from this location.

Reply:
IMHO "native" is correct here. We are using the same name elsewhere in the image validator as well.

Reply:
@superm1 something like the fixup commit?